AI/VCAI/BuildingManager.cpp

```diff
@@ -172,33 +173,42 @@ bool BuildingManager::getBuildingOptions(const CGTownInstance * t)
 	if(tryBuildAnyStructure(t, essential))
 		return true;
 
-	//the more gold the better and less problems later //TODO: what about building mage guild / marketplace etc. with city hall disabled in editor?
-	if(tryBuildNextStructure(t, basicGoldSource))
-		return true;
-
-	//workaround for mantis #2696 - build capitol with separate algorithm if it is available
-	if(vstd::contains(t->builtBuildings, BuildingID::CITY_HALL) && getMaxPossibleGoldBuilding(t) == BuildingID::CAPITOL)
+	if (cb->getDate(Date::DAY_OF_WEEK) < 5) // first 4 days of week - try to focus on dwellings
 	{
-		if(tryBuildNextStructure(t, capitolAndRequirements))
+		if (tryBuildNextStructure(t, unitsSource, 4))
 			return true;
 	}
 
-	if(!t->hasBuilt(BuildingID::FORT)) //in vast majority of situations fort is top priority building if we already have city hall, TODO: unite with unitGrowth building chain
-		if(tryBuildThisStructure(t, BuildingID::FORT))
+	if (cb->getDate(Date::DAY_OF_WEEK) > 4) // last 3 days of week - try to focus on growth by building Fort/Citadel/Castle
+	{
+		if (tryBuildNextStructure(t, defence, 3))
 			return true;
+	}
 
+	if (t->hasBuilt(BuildingID::CASTLE))
+	{
+		if (tryBuildAnyStructure(t, unitGrowth, 4))
+			return true;
+	}
 
+	//make required structures for City Hall
+	if(tryBuildNextStructure(t, cityhallRequirements))
+		return true;
```

Comment:
Is this necessary? tryBuildNextStructure builds requirements first, so code like `tryBuildThisStructure(t, BuildingID::CITY_HALL)` should build Marketplace/Blacksmith/Mages Guild anyway

Reply:
Just being thorough in specifying what needs to be built since older code comments indicated that sometimes requirements were not being built when expected.

Reply:
Such comments might be outdated.

And if such a bug exists then it should be investigated.
Sure, you've added fix for City Hall. But what if such bug triggers with Tavern and Town Hall?
Or with any dwelling which might have different requirements, including towns from mods?